docs/source/architecture/data_services_architecture.rst

```diff
@@ -1,4 +1,110 @@
+
+    * Csv Import Export Customizer: Service providing methods for importing and exporting instances using csv.
+
+    * Motech Data Service: Centre of Data Access Layer is Generic Data Access Pattern Class which defines generic persistence related methods. Superclass for defining service of any Entity in MDS.
+
+    * Entity Service: It defines CRUD actions on entities.
+
+    * History Service: It persists changes made on given instance. It captures instances which are not in trash.
+
+    * Trash Service: It provides methods related to trash mode (By Default, trash mode is active and can be turned off by user.)
+
+    * Jar Generator Service: It creates a bundle jar with all entities defined in MDS.
+
+    * Json LookUp Service: Service for managing lookups from JSON.
+
+    * MDS Bundle Regeneration Service: It provides methods to regenerate bundle and regenerate bundle after dde enhancement. This service use Jar Generator Service to regenerate bundle.
+
+    * MDS LookUp Service: It provides a methods for processing lookups for particular entity given names, class names as well as String names. It provides a generic access to entities.
+
+    * MDS Scheduler Service: It provides methods to schedule and unschedule a repeating job. It used for emptying trash periodically.
+
+    * Migration Service: It provides method to find flyway migrations within bundle (Default Search location is db/migration).
+
+    * Type Service: It provides methods for defining methods for executing actions on type.
+    Rest Documentation Service. It provides functionality of dynamically generated MDS Rest Documentation.
+
+    * JDO Listener Registry Service: It provides access to registry of Listeners for Persistence events .i.e. POST_CREATE, PRE_DELETE, POST_DELETE, POST_LOAD, PRE_STORE and POST_STORE.It handles pre and post events of crud events of DataNuclues.
+
+
+
+
+Notes
+-----
+*  Custom Annotations, their processing and actions performed on those annotations is a part of MDS. These processors process DDEs and create Metadata as well as Infrastructure Information for those entities which can used via DataNucleus enhancer to enhance entities to use them for Persistence Operations.
+*  EUDEs, the related Data Access Layer and their related MetaData Information is generated which can be used by DataNucleus for Persistence Operations.
+
+*  Above mentioned Entities and related Data Access Layer and Metadata information are created during bootstrapping of MOTECH Platform.
+
+*  Whenever an EUDE is defined from UI, 'MOTECH Data Services' bundle is regenerated.
+
+*  Whenever EUDE lookups are defined using UI, 'MOTECH Data Service' bundles is regenerated.
+
+* MDS also provides features to create REST APIs for creating, updating, deleting and searching the instances of the Entities created.
+
+* Feature To download/upload schema using csv as input or output.
```

Comment:
This doc seems to lack information about the databases being used - MDS uses two databases now.